lib/deflation.cpp

```diff
@@ -22,6 +23,9 @@ namespace quda {
   using DynamicStride = Stride<Dynamic, Dynamic>;
 #endif
 
+  static auto allocator = [=] (size_t bytes ) { return static_cast<Complex*>(mapped_malloc(bytes)); };
```

Comment:
This doesn't compile with clang. Seems like gcc is more permissive here. Can this be fixed?

```
error: non-local lambda expression cannot have a capture-default [clang-diagnostic-error]
  static auto allocator = [=] (size_t bytes ) { return static_cast<Complex*>(mapped_malloc(bytes)); };
                           ^
Suppressed 1 warnings (1 in non-user code).
```